samples/cpp/videocapture_depth.cpp

```diff
@@ -0,0 +1,175 @@
+        if (camType == "OpenNI")
+        {
+            backend = CAP_OPENNI2;
+            flagDepth = CAP_OPENNI_DEPTH_MAP;
+            flagBGR = CAP_OPENNI_BGR_IMAGE;
+            flagIR = CAP_OPENNI_IR_IMAGE;
+            hasDisparity = true;
+        }
+        else if (camType == "RealSense")
+        {
+            backend = CAP_INTELPERC;
+            flagDepth = CAP_INTELPERC_DEPTH_MAP;
+            flagBGR = CAP_INTELPERC_IMAGE;
+            flagIR = CAP_INTELPERC_IR_MAP;
+        }
+        else if (camType == "Orbbec")
+        {
+            backend = CAP_OBSENSOR;
+            flagDepth = CAP_OBSENSOR_DEPTH_MAP;
+            flagBGR = CAP_OBSENSOR_BGR_IMAGE;
+            flagIR = CAP_OBSENSOR_IR_IMAGE;  // Note output IR stream is not implemented for now.
+        }
+        else
+        {
+            std::cerr << "Invalid input of camera type." << std::endl;
+            return -1;
+        }
+
+        VideoCapture capture(backend);
+        if(!capture.isOpened())
+        {
+            std::cerr << "Fail to open depth camera." << std::endl;
+            return -1;
+        }
+
+        if (camType == "OpenNI")
+        {
+            capture.set(CAP_OPENNI_IMAGE_GENERATOR_OUTPUT_MODE, CAP_OPENNI_VGA_30HZ);
+            capture.set(CAP_OPENNI_DEPTH_GENERATOR_PRESENT, true);
+            capture.set(CAP_OPENNI_IMAGE_GENERATOR_PRESENT, true);
+            capture.set(CAP_OPENNI_IR_GENERATOR_PRESENT, true);
+        }
+
+        Mat depthMap;
+        Mat bgrImage;
+        Mat irImage;
+        Mat disparityMap;
+
+        for(;;)
+        {
+            if(capture.grab())
+            {
+                if(capture.retrieve(depthMap, flagDepth))
+                {
+                    Mat colorDepthMap;
+                    colorizeDepth(depthMap, colorDepthMap);
+                    imshow("Colorized Depth Map", colorDepthMap);
+                }
+
+                if(capture.retrieve(bgrImage, flagBGR))
+                    imshow("RGB Image", bgrImage);
+
+                if(capture.retrieve(irImage, flagIR))
+                {
+                    if (camType == "OpenNI")
+                    {
+                        Mat ir8;
+                        irImage.convertTo(ir8, CV_8U, 256.0 / 3500, 0.0);
+                        imshow("Infrared Image", ir8);
+                    }
+                    else
+                        imshow("Infrared Image", irImage);
+                }
+
+                if (hasDisparity)
+                {
+                    if(capture.retrieve(disparityMap, CAP_OPENNI_DISPARITY_MAP))
+                    {
+                        Mat colorDisparityMap;
+                        colorizeDisparity(disparityMap, colorDisparityMap, getMaxDisparity(capture));
+                        Mat validColorDisparityMap;
+                        colorDisparityMap.copyTo(validColorDisparityMap, disparityMap != 0);
+                        imshow("Colorized Disparity Map", validColorDisparityMap);
```

Comment:
Invalid pixel values are unpredictable here, because `validColorDisparityMap` is not initialized. I propose not create `validColorDisparityMap`, but call setTo with mask for for `colorDisparityMap` to mark invalid pixels with black.